x-pack/plugin/core/src/main/java/org/elasticsearch/license/License.java

```diff
@@ -469,9 +470,17 @@ public static License fromXContent(XContentParser parser) throws IOException {
         }
         // not a license spec
         if (builder.signature != null) {
-            byte[] signatureBytes = Base64.getDecoder().decode(builder.signature);
-            ByteBuffer byteBuffer = ByteBuffer.wrap(signatureBytes);
-            int version = byteBuffer.getInt();
+            int version;
+            // In case the signature is truncated/malformed we might end up with fewer than 4 bytes in the byteBuffer
+            // or with a string that cannot be base64 decoded. In either case return a more friendly error instead of
+            // just throwing the BufferUnderflowException or the IllegalArgumentException
+            try {
+                byte[] signatureBytes = Base64.getDecoder().decode(builder.signature);
+                ByteBuffer byteBuffer = ByteBuffer.wrap(signatureBytes);
+                version = byteBuffer.getInt();
+            } catch (BufferUnderflowException | IllegalArgumentException e) {
```

Comment:
Is it intentional that you don't just catch `Exception`? I don't think there's any exception that could be thrown here that we wouldn't want to wrap with a better message.

Reply:
Yes, I did it on purpose. `decode()` throws an `IllegalArgumentException` and `getInt()` throws a `BufferUnderflowException()` . I think that `wrap()` can't throw an `IndexOutOfBoundsException` when called with a byte array param. Am I missing any ?

Reply:
I think it is safe to catch `Exception` here. Any exception that we get here would pretty much indicate a malformed signature right?

Reply:
Yeap, I'll adjust this